cypress/page/t-shirts/t-shirts.page.ts

```diff
@@ -8,12 +8,19 @@ export class Tshirts{
       this.addToCartButton = '.ajax_add_to_cart_button.btn.btn-default';
     }
   
+
+    public findTShirtByName(tshirtName:string){
+    
+      return cy.get(".product-name").filter(`[title="${tshirtName}"]`);
+      
+     }
     public selectTshirt(tshirtName:string) {
-      cy.get(this.tshirts)
-        .find('.product-name')
+        this.findTShirtByName(tshirtName)
         .contains(tshirtName)
         .first()
         .get(this.addToCartButton)
```

Comment:
we can avoid this if you use again filter or find method using the addToCartButton selector